integrations/sonarqube/client.py

```diff
@@ -284,82 +263,115 @@ async def get_single_project(self, project: dict[str, Any]) -> dict[str, Any]:
 
         return project
 
-    async def get_all_projects(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+    async def _get_projects(
+        self, params: dict[str, Any]
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for projects in self._handle_paginated_request(
+            endpoint=Endpoints.PROJECTS,
+            data_key="components",
+            method="GET",
+            query_params=params,
+        ):
+            yield projects
```

Comment:
why do we need this intermediate method for fetching projects?

Reply:
To ensure separation of concerns, else the `get_all_projects` method will be doing too much